asio/include/asio/detail/is_buffer_sequence.hpp

```diff
@@ -214,8 +220,8 @@ char mutable_buffers_type_typedef_helper(
 template <typename T, typename Buffer>
 struct is_buffer_sequence_class
   : integral_constant<bool,
-      sizeof(buffer_sequence_begin_helper<T>(0)) != 1 &&
-      sizeof(buffer_sequence_end_helper<T>(0)) != 1 &&
+      sizeof(buffer_sequence_begin_helper<T>(0, 0)) != 1 &&
+      sizeof(buffer_sequence_end_helper<T>(0, 0)) != 1 &&
```

Comment:
Now the fallback becomes `char[2]` not `char, so should it not be `!=2`

Reply:
When called with single argument this is revolved to the fallback, since no the default argument value in checker function declaration is specified. The modified fallback for `ASIO_HAS_DECLTYPE` case is `char`, so `!= 1` is correct.

When `ASIO_HAS_DECLTYPE` is not defined, fallback is still `char[2]`, but in this case checker function must be rejected for acceptable type; `!= 1` is correct again.